src/main/webapp/app/course/manage/course-detail.component.ts

```diff
@@ -62,16 +128,52 @@ export class CourseDetailComponent implements OnInit, OnDestroy {
      */
     ngOnDestroy() {
         this.eventManager.destroy(this.eventSubscriber);
+        this.dialogErrorSource.unsubscribe();
+        this.websocketService.unsubscribe(CourseDetailComponent.getCourseArchiveStateTopic(this.course.id!));
     }
 
-    /**
-     * Subscribe to changes in courses and reload the course after a change.
-     */
-    registerChangeInCourses() {
-        this.eventSubscriber = this.eventManager.subscribe('courseListModification', () => {
-            this.courseService.find(this.course.id!).subscribe((courseResponse: HttpResponse<Course>) => {
-                this.course = courseResponse.body!;
-            });
-        });
+    private static getCourseArchiveStateTopic(courseId: number) {
+        return '/topic/courses/' + courseId + '/archive-course';
+    }
+
+    canArchiveCourse() {
+        // A course can only be archived if it's over.
+        const isCourseOver = this.course.endDate?.isBefore(moment()) ?? false;
+        return !!this.course.isAtLeastInstructor && isCourseOver;
+    }
+
+    archiveCourse() {
+        this.courseService.archiveCourse(this.course.id!).subscribe();
+    }
+
+    downloadCourseArchive() {
+        this.courseService.downloadCourseArchive(this.course.id!).subscribe(
+            (response) => downloadZipFileFromResponse(response),
+            () => this.jhiAlertService.error('artemisApp.course.archive.archiveDownloadError'),
+        );
+    }
+
+    canDownloadArchive() {
+        const hasArchive = !!this.course.courseArchivePath && this.course.courseArchivePath.length > 0;
+        // You can only download one if the path to the archive is present
+        return !!this.course.isAtLeastInstructor && hasArchive;
```

Comment:
No need for !! Since isAtLeastInstructor is already a boolean, right?

Reply:
Sadly no, it can also be undefined so I convert it into a boolean like this

Reply:
👍